crates/bevy_ecs/src/system/system.rs

```diff
@@ -66,14 +67,15 @@ pub(crate) fn check_system_change_tick(
     change_tick: u32,
     system_name: &str,
 ) {
-    let tick_delta = change_tick.wrapping_sub(*last_change_tick);
-    const MAX_DELTA: u32 = (u32::MAX / 4) * 3;
-    // Clamp to max delta
-    if tick_delta > MAX_DELTA {
+    let delta = change_tick.wrapping_sub(*last_change_tick);
+    // This comparison assumes that `delta` has not overflowed `u32::MAX` before, which will be true
+    // so long as a check runs at least every 2 * `CHANGE_DETECTION_CHECK_THRESHOLD` ticks.
+    if delta > CHANGE_DETECTION_MAX_DELTA {
         warn!(
-            "Too many intervening systems have run since the last time System '{}' was last run; it may fail to detect changes.",
-            system_name
+            "{} intervening systems have run since system '{}' last ran. \
```

Comment:
IMO we should be more specific about the failure mode here.

Reply:
So once a change ages to `CHANGE_DETECTION_MAX_DELTA`, `ticks_since_system > ticks_since_change` always returns `false`. Basically, every change "expires" after `CHANGE_DETECTION_MAX_DELTA` ticks.

Is something like that enough detail?

Reply:
Yep! IMO something like this should be enough for the end user.

"Changes from longer than CHANGE_DETECTION_MAX_DELTA ticks may be missed."